### PR TITLE
Fix issue with collective metadata writes of global heap data

### LINKED ISSUE #2433: [BUG] NetCDF fails its parallel tests with HDF 1.14.0
**Describe the bug**
The NetCDF parallel tests fail due to an assertion in H5F_shared_vector_write (H55Fioc).

**Expected behavior**
NetCDF Parallel tests should not fail.

**Platform (please complete the following information)**
 - HDF5 version (if building from a maintenance branch, please include the commit hash)
 - OS and version
 - Compiler and version
 - Build system (e.g., CMake, Autotools) and version
 - Any configure options you specified
 - MPI library and version (parallel HDF5)

tested  HDF5 13.1 (passes), 13.2(fails), 14.0(fails) and develop (fails).
Linux, GCC 11, MPICH 4, Autotools, NetCDF 4.9.1.

nc_create_par will fail assertion:
../../src/H5Fio.c:397: H5F_shared_vector_write: Assertion `types[i] != H5FD_MEM_GHEAP' failed.

All the NetCDF tests pass when the assert is removed.

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Fixes #2433

### CODE REVIEW

**Review by brtnfld — APPROVED** [review]
LGTM, I verified that it fixes the NetCDF failures 🔧

**Review by qkoziol — CHANGES_REQUESTED** [review]
lgtm, but should have a new regression test also

**Comment by jhendersonHDF** [comment]
> lgtm, but should have a new regression test also

I agree we need a reproducer, though it might be difficult to come up with one. Any idea how a parallel netCDF program is causing a write of global heap metadata? My only initial thought was reference and/or VL types, but we don't support access to the global heap in parallel.

**Comment by qkoziol** [comment]
VL or ref datatype in an attribute?

**Comment by jhendersonHDF** [comment]
> VL or ref datatype in an attribute?

Ah ok.. I suppose that wouldn't really be problematic parallel access since all ranks would be doing the same thing.

**Review by qkoziol — APPROVED** [review]
Great!

**Review by lrknox — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -207,6 +207,21 @@ Bug Fixes since HDF5-1.13.3 release
 ===================================
     Library
     -------
+    - Fixed an issue with collective metadata writes of global heap data
```
Comment (wkliao) [inline release_docs/RELEASE.txt:210]: Shouldn't this statement be under "Bug Fixes since HDF5-1.14.0 release", instead of 1.13.3?

**Pull request merged**